src/main/java/frc/robot/posev2/RobotPoseManager.java

```diff
@@ -0,0 +1,44 @@
+  
+package frc.robot.posev2;
+
+import frc.robot.RobotMap;
+
+
+public class RobotPoseManager {
+ 
+    EncoderValues encoders;
+    NavXData navXData;
+    VisionData vData;
+    WheelColorValue wColor;
+
+    RobotPose pose = RobotMap.DIMENSIONS.START_POSE;
+    private boolean navXWorking = true;
+    
+    public RobotPose getCurrentPose(){
+        return pose;
+    }
+    
+    public void update(){
+        //do all the maths to get a new pose
+        pose = PoseMathematics.addPoses(pose, PoseMathematics.calculateRobotPositionChange(encoders.getDeltaLeft(), encoders.getDeltaRight()));
+        if(navXWorking){
```

Comment:
i dont think you want to throw away all of the pose update, just becaues the navx isnt working, right? probaly if the navx isnt working, you want to still populate the other values?